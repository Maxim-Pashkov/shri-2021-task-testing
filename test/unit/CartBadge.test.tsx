import { it, expect, describe } from '@jest/globals';
import { render } from '@testing-library/react';
import { getApplication } from './getApplication';
import { CartBadge } from '../../src/client/components/CartBadge';

const cartState = {
    5: {
        name: 'test',
        price: 10,
        count: 1,
    },
};

describe('CartBadge test', () => {
    it('if an item is in the cart, a badge will be displayed', () => {
        const { application } = getApplication(
            () => (
                <div data-testid="badge">
                    <CartBadge id={5} />
                </div>
            ),
            cartState
        );

        const { container } = render(application);

        expect(
            !!container.querySelector('[data-testid="badge"]').childElementCount
        ).toBeTruthy();
    });

    it("if an item is't in the cart, a badge will't be displayed", () => {
        const { application } = getApplication(
            () => (
                <div data-testid="badge">
                    <CartBadge id={6} />
                </div>
            ),
            cartState
        );

        const { container } = render(application);
        expect(
            !!container.querySelector('[data-testid="badge"]').childElementCount
        ).toBeFalsy();
    });
});
