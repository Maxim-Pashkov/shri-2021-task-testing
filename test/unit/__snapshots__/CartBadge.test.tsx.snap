// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartBadge test if an item is in the cart, a badge will be displayed 1`] = `
<span
  class="CartBadge text-success mx-3"
>
  Item in cart
</span>
`;

exports[`CartBadge test if an item is't in the cart, a badge will't be displayed 1`] = `null`;
